src/backend/EinsumNode.cpp

```diff
@@ -327,20 +336,50 @@ einsum_ir::err_t einsum_ir::backend::EinsumNode::compile() {
     m_offset_bytes *= ce_n_bytes( m_dtype );
   }
 
-  // compile children
+  // set required memory
+  if( m_data_ptr_ext == nullptr || l_permute_inputs ) {
+    m_req_mem = m_size;
+  }
+  else{
+    m_req_mem = 0;
+  }
+
+  // compile children and determine best execution order
   if( m_children.size() > 1 ) {
     for( std::size_t l_ch = 0; l_ch < m_children.size(); l_ch++ ) {
-      l_err = m_children[l_ch]->compile();
+      l_err = m_children[l_ch]->compile_recursive();
       if( l_err != einsum_ir::SUCCESS ) {
         return l_err;
       }
     }
+    int64_t m_mem_ch1 = m_children[0]-> m_mem_subtree;
+    int64_t m_mem_ch2 = m_children[1]-> m_mem_subtree;
+
+    int64_t m_max_ch1 = std::max(m_mem_ch1, m_mem_ch2 + m_children[0]->m_req_mem);
+    int64_t m_max_ch2 = std::max(m_mem_ch2, m_mem_ch1 + m_children[1]->m_req_mem);
```

Comment:
``m_`` prefix is reserved for class members. Use ``l_`` here since its in the local scope.